core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -404,14 +404,8 @@ class LogValidatorTest {
     assertEquals(now + 1, validatingResults.maxTimestampMs,
       s"Max timestamp should be ${now + 1}")
 
-    val expectedShallowOffsetOfMaxTimestamp = if (magic >= RecordVersion.V2.value) {
```

Comment:
if it's LOG_APPEND_TIME, it should be the first record. If not, then we should choose the correct one